File: exs/mundo_1/c#/026.cs
/*
Desafio 026

Problema: Faça um programa que leia uma frase pelo teclado
          e mostre quantas vezes aparece a letra "A", em que
          posição ela aparece a primeira vez e em que posição
          ela aparece a última vez.

Resolução do problema:
*/

using System;

class io {
    public static string input(string text) {
        Console.Write(text);
        return Console.ReadLine();
    }
    
    public static void print(object data, string end="\n") {
        string text = data.ToString();

        Console.Write($"{text}{end}");
    }
}

class Program {
  public static void Main () {
    string frase = io.input("Digite uma frase qualquer: ");
    string lowerFrase = frase.ToLower();

    int primeiraOcorrencia = lowerFrase.IndexOf('a');
    int ultimaOcorrencia = lowerFrase.LastIndexOf('a');

    int lengthFrase = lowerFrase.Length;
    int lengthFraseSemA = lowerFrase.Replace("a", "").Length;
    int lengthA = (lengthFrase - lengthFraseSemA);

    io.print($"Quantas vezes a letra \"A\" apareceu: {lengthA}");
    io.print($"Ela aparece a primeira vez na posição: {primeiraOcorrencia + 1}");  
    io.print($"Ela aparece a ultima vez na posição: {ultimaOcorrencia + 1}");
  }
}

File: exs/mundo_1/java/026.java
/*
    ### 026 - Primeira e última ocorrência de uma string
    Faça um programa que leia uma frase qualquer e mostre: quantas vezes aparece a letra "a",
    em que posição ela aparece a primeira vez em que posição ela aparece a última vez
 */


import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        String frase;
        int contador = 0;
        Scanner scanner = new Scanner(System.in);

        System.out.println("Digite uma frase qualquer: ");
        frase = scanner.nextLine();

        for (int i = 0; i < frase.length(); i++)
        {
            if (frase.toCharArray()[i] == 'a') contador++;
        }
        System.out.println("a apareceu: " + contador + " vezes");
        System.out.println("Posicao da primeira vez: " + frase.indexOf("a"));
        System.out.println("Posicao da ultima vez: " + frase.lastIndexOf("a"));
    }

}
